### PR TITLE
refractor service worker

### PR DESCRIPTION
**Pull request by chenxsan**
We have `webpack.ssg.js` to generate those static html pages, but the current service worker is not taking full advantage of them.

For example, when we open `view-source:https://webpack.js.org/guides/`, we expect `dist/guides/index.html` to be returned. Instead `/app-shell/index.html` is returned at the moment.

This pull request refractors the service worker replacing `GenerateSW ` with `InjectManifest` which gives us more control.

- [x] add some e2e tests

### CODE REVIEW

**Comment by chenxsan** [comment]
Got a weird issue in CI https://travis-ci.org/github/webpack/webpack.js.org/jobs/741364115#L283:

```
setup compilation HtmlWebpackPluginTypeError: Cannot read property 'Compilation' of undefined
    at compiler.hooks.thisCompilation.tap (/home/travis/build/webpack/webpack.js.org/node_modules/html-webpack-plugin/index.js:210:26)
    at Hook.eval [as call] (eval at create (/home/travis/build/webpack/webpack.js.org/node_modules/tapable/lib/HookCodeFactory.js:19:10), <anonymous>:64:1)
    at Hook.CALL_DELEGATE [as _call] (/home/travis/build/webpack/webpack.js.org/node_modules/tapable/lib/Hook.js:14:14)
    at Compiler.newCompilation (/home/travis/build/webpack/webpack.js.org/node_modules/webpack/lib/Compiler.js:915:30)
    at hooks.beforeCompile.callAsync.err (/home/travis/build/webpack/webpack.js.org/node_modules/webpack/lib/Compiler.js:957:29)
    at Hook.eval [as callAsync] (eval at create (/home/travis/build/webpack/webpack.js.org/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:40:1)
    at Hook.CALL_ASYNC_DELEGATE [as _callAsync] (/home/travis/build/webpack/webpack.js.org/node_modules/tapable/lib/Hook.js:18:14)
    at Compiler.compile (/home/travis/build/webpack/webpack.js.org/node_modules/webpack/lib/Compiler.js:952:28)
    at compiler.hooks.watchRun.callAsync.err (/home/travis/build/webpack/webpack.js.org/node_modules/webpack/lib/Watching.js:113:19)
    at Hook.eval [as callAsync] (eval at create (/home/travis/build/webpack/webpack.js.org/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:26:1)
```

File: package.json [hunk 1]
```diff
@@ -57,7 +57,7 @@
     "jest": "jest",
     "cypress:open": "cypress open",
     "cypress:run": "cypress run",
-    "cypress:ci": "start-server-and-test http-get://localhost:3000 cypress:run"
+    "cypress:ci": "start-server-and-test build-test http://localhost:4200 cypress:run"
```
Comment (chenxsan) [inline package.json:60]: We're testing against the production site now, I think it makes more sense.

**Comment by chenxsan** [comment]
Oops, we have a problem https://github.com/GoogleChrome/workbox/issues/2493 now:

> Uncaught ReferenceError: regeneratorRuntime is not defined

File: src/PrecacheSsgManifestPlugin.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
+// we need to precache some assets from ssg too
+// they're previously handled by require('./src/utilities/find-files-in-dist')(['.css', '.ico', '.svg'])
+
+const { Compilation, sources } = require('webpack');
+const getManifestEntriesFromCompilation = require('workbox-webpack-plugin/build/lib/get-manifest-entries-from-compilation');
+
+module.exports = class PrecacheSsgManifestPlugin {
```
Comment (montogeek) [inline src/PrecacheSsgManifestPlugin.js:7]: Please move this to an external dependency.

**Pull request merged**